patroni/api.py

```diff
@@ -556,7 +556,12 @@ def __httpserver_init(self, host, port):
         info.sort(key=lambda x: x[0] == socket.AF_INET, reverse=not dual_stack)
 
         self.address_family = info[0][0]
-        HTTPServer.__init__(self, info[0][-1][:2], RestApiHandler)
+        try:
+            HTTPServer.__init__(self, info[0][-1][:2], RestApiHandler)
+        except OSError:
+            raise PatroniException(
```

Comment:
1. Now we are hiding the real error, `Address already in use` or something like this. It makes it much harder to figure out what it wrong.
2. It would be nice to add the test